pandas/tests/groupby/test_apply.py

```diff
@@ -974,3 +974,21 @@ def test_apply_function_with_indexing_return_column():
     result = df.groupby("foo1", as_index=False).apply(lambda x: x.mean())
     expected = DataFrame({"foo1": ["one", "three", "two"], "foo2": [3.0, 4.0, 4.0]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_apply_function_called_count(capsys):
+    # GH: 31111
+    # groupby-apply need to execute len(set(group_by_columns)) times
+    # `https://github.com/pandas-dev/pandas/issues/31111`
```

Comment:
If you already have the issue number you don't need the link as well

Reply:
Alright. I understood that. Will make sure not to repeat this again.

Reply:
Changed accordingly. @MarcoGorelli